### ISSUE TITLE
Bizzare bug with helm chart

### ISSUE THREAD

**Issue by mateo-moon** [joshstevens19/rindexer#136]
Hey!
I've got very bizzare bug here. Using the repo chart i deployed it substituting rindexer.yaml and abi content in config maps. Also there is a postgres deployment by bitnami chart.
After the pod starts the rindexer prints this logs:

```Logger has already been set up, continuing...
30 December - 13:58:48.830055  INFO Starting rindexer no code
30 December - 13:58:48.830157  INFO Setting up postgres
30 December - 13:58:48.913451  INFO Creating tables for ProofDotFun
30 December - 13:58:48.913689  INFO Creating schema if not exists: proof_dot_fun_proof_dot_fun
30 December - 13:58:48.913726  INFO Creating table if not exists: proof_dot_fun_proof_dot_fun.beneficiary_claimed
30 December - 13:58:48.913753  INFO Creating table if not exists: proof_dot_fun_proof_dot_fun.token_created
30 December - 13:58:48.913773  INFO Creating table if not exists: proof_dot_fun_proof_dot_fun.token_graduated
30 December - 13:58:48.913782  INFO Creating table if not exists: proof_dot_fun_proof_dot_fun.token_transaction
30 December - 13:58:48.926194  INFO Created tables for ProofDotFun
30 December - 13:58:48.977610  INFO Networks enabled: base-sepolia
30 December - 13:58:48.978136  INFO Events registered to index:ProofDotFun::BeneficiaryClaimed, ProofDotFun::TokenCreated, ProofDotFun::TokenGraduated, ProofDotFun::TokenTransaction
30 December - 13:58:48.979122  INFO Temp dropping constraints relationships from the database for historic indexing for speed reasons
30 December - 13:58:48.979160  INFO Starting GraphQL server
```

After that it stalls. Nothing happens.
And here is the most interesting thing. If i run `kubectl -n rindexer debug -ti rindexer --target rindexer --image=ghcr.io/joshstevens19/rindexer-bundled:latest -- bash` copy config rindexer.yaml and abi directory from /proc/1/root (which is basically main container root) to the temp of ephemeral container and run `/app/rindexer start all` it runs FLAWLESSLY) But when i do the same with ` kubectl -n rindexer exec -ti rindexer bash` and try to run the same it stalls at the same point as above and after i quit process with CTR+C it prints this: 

```
ERROR GraphQL: Could not start up API: Child process exited with errors
14:53:02  INFO Received SIGINT (Ctrl+C) signal gracefully shutting down...
14:53:02  INFO Starting shutdown with 0 active tasks
14:53:02  INFO All active indexing tasks finished shutting down system...
14:53:02  INFO Shutdown complete
```

I've tried different set ups within k8s cluster. Different nodes, security contexts, namespaces, files(i've downloaded config files directly not using config map binds) and whatnot. `ps auxf` returns the same in both containers, df -h returns the same. My head hurts.... Help me please)

**Comment by joshstevens19** [comment]
@absis is going to take a look as he did the helm - I think maybe we need to use rindexer-bundled image everywhere as the one without it doesn’t have the graphql exe in it and maybe why it crashed here and your other command works

**Comment by mateo-moon** [comment]
> @absis is going to take a look as he did the helm - I think maybe we need to use rindexer-bundled image everywhere as the one without it doesn’t have the graphql exe in it and maybe why it crashed here and your other command works

Hey, Josh. Thx for your reply. I didn't get you. There are no difference in what binaries i used. I used only rindexer from oficial image rindexer-bundled. And the command to run is always the same as well `/app/rindexer start all`

**Comment by mateo-moon** [comment]
I've been writing big tldr with commands and examples based on the original chart and during this process i also tested this behavior again. And found that the error was actually different of what i got with my variation. I took a closer look at diff and noticed that the original chart got this resources:
```
 resources:
  limits:
    cpu: 100m
    memory: 128Mi
  requests:
    cpu: 100m
    memory: 128Mi

```

And mine is:

```
resources:
  limits:
    cpu: 800m
    memory: 4Gi
  requests:
    cpu: 400m
    memory: 500Mi
```

That got me thinking and i removed it completely and you know what) It started without problems)
I've checked the actual resource consumption and it turns out that pod itself uses around 50 mb but apparently initializing step needs more.
<img width="1366" alt="Screenshot 2024-12-31 at 12 08 09" src="https://github.com/user-attachments/assets/9193f8cf-88a6-43ba-a119-0a5ae3923e03" />
I ran some tests and found out that the threshold somewhere between 500Mi and 600Mi. I was unlucky enough to set requests for 500Mi slightly lower than it needs.🤦‍♂️
Also you can notice that the limits are actually much higher, but apparently it doesn't matter in this case 🤷‍♂️. Interesting to know how this works under the hood.
Anyway, this thread helped me to solve this problem. Thx guys.

**Comment by joshstevens19** [comment]
Oh this is interesting we should adjust our to be the bare minimum to work out the box 

https://github.com/joshstevens19/rindexer/blob/master/helm/rindexer/values.yaml#L20

maybe we can adjust requests::memory > 600Mi 

as you identified this even though we expect people to adjust their end happy accept a PR for this @mateo-moon if you wanted a contribution let me know

Glad it’s all sorted great finding we can probably improve the docs to explain this again happily accept a PR on that finding @mateo-moon

**Comment by mateo-moon** [comment]
I'm glad to helm with such a great project) Here is fix - #138

**Comment by joshstevens19** [comment]
Amazing left a comment on docs then we can merge closing this thanks a lot
